src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -777,65 +841,22 @@ static void check_signal_handler(int sig) {
 
   os_sigaction(sig, (struct sigaction*)NULL, &act);
 
-  // See comment for SA_RESTORER_FLAG_MASK
-  LINUX_ONLY(act.sa_flags &= SA_RESTORER_FLAG_MASK;)
-
-  address thisHandler = get_signal_handler(&act);
-
-  switch (sig) {
-  case SIGSEGV:
-  case SIGBUS:
-  case SIGFPE:
-  case SIGPIPE:
-  case SIGILL:
-  case SIGXFSZ:
-    jvmHandler = CAST_FROM_FN_PTR(address, (sa_sigaction_t)javaSignalHandler);
-    break;
-
-  case SHUTDOWN1_SIGNAL:
-  case SHUTDOWN2_SIGNAL:
-  case SHUTDOWN3_SIGNAL:
-  case BREAK_SIGNAL:
-    jvmHandler = (address)os::user_handler();
-    break;
-
-  default:
-    if (sig == PosixSignals::SR_signum) {
-      jvmHandler = CAST_FROM_FN_PTR(address, (sa_sigaction_t)SR_handler);
-    } else {
-      return;
-    }
-    break;
-  }
-
-  if (thisHandler != jvmHandler) {
-    tty->print("Warning: %s handler ", os::exception_name(sig, buf, O_BUFLEN));
-    tty->print_raw("expected:");
-    print_signal_handler_name(tty, jvmHandler, buf, O_BUFLEN);
-    tty->print_raw("  found:");
-    print_signal_handler_name(tty, thisHandler, buf, O_BUFLEN);
-    // No need to check this sig any longer
-    sigaddset(&check_signal_done, sig);
+  // Compare both sigaction structures (intelligently; only the members we care about).
+  if (!compare_handler_info(&act, expected_act)) {
+    tty->print_cr("Warning: %s handler modified!", os::exception_name(sig, buf, sizeof(buf)));
+    // If we had a mismatch:
+    // - print all signal handlers. As part of that printout details will be printed
```

Comment:
Needs comma:

`// - print all signal handlers. As part of that printout, details will be printed`